scripts/mysql_system_tables_fix.sql

```diff
@@ -1028,23 +1035,87 @@ ALTER TABLE help_relation CONVERT TO CHARACTER SET utf8mb3;
 ALTER TABLE help_keyword CONVERT TO CHARACTER SET utf8mb3;
 
 --
--- Upgrade a table engine from MyISAM to InnoDB for the system tables
+-- Upgrade a table engine from MyISAM to MyRocks or InnoDB for the system tables
 -- help_topic, help_category, help_relation, help_keyword, plugin, servers,
 -- time_zone, time_zone_leap_second, time_zone_name, time_zone_transition,
 -- time_zone_transition_type, columns_priv, db, procs_priv, proxies_priv,
 -- tables_priv, user.
-ALTER TABLE func ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE help_topic ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE help_category ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE help_relation ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE help_keyword ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE plugin ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE servers ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE time_zone ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE time_zone_leap_second ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE time_zone_name ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE time_zone_transition ENGINE=InnoDB STATS_PERSISTENT=0;
-ALTER TABLE time_zone_transition_type ENGINE=InnoDB STATS_PERSISTENT=0;
+SET @cmd = CONCAT("ALTER TABLE func ENGINE=",
```

Comment:
why missing some of system tables? such as mysql.component, mysql.default_roles..?

Reply:
I believe this section deals with the old instance upgrade where those tables have to be converted from MyISAM. For our use case, the InnoDB -> MyRocks conversion is not handled here.

Reply:
okay.. sounds good.. I plan to also use this script(instead of code) to handle InnoDB -> MyRocks system tables.